plugins/node/opentelemetry-instrumentation-cucumber/src/instrumentation.ts

```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { diag, Span, SpanKind, SpanStatusCode } from '@opentelemetry/api';
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  InstrumentationNodeModuleFile,
+  isWrapped,
+} from '@opentelemetry/instrumentation';
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+
+import type * as cucumber from '@cucumber/cucumber';
+import * as messages from '@cucumber/messages';
```

Comment:
https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1252/commits/9170a5aab58636ff3c0d8d09441a220243ec5979
Is this what you had in mind?

Reply:
Yes :)
One last suggestion. Now these texts are a bit magical
```
if (['UNDEFINED', 'AMBIGUOUS', 'FAILED'].includes(status)) {
```

I imagine it could increase readability and maintainability if instead, we copy the enum into the instrumentation code and use it as enum.

something like:
```js
// internal-types.ts

// copied from instrumented package
//https://github.com/cucumber/messages/blob/1b570700edc26a1764e28c0043ef1126d9db0cbb/javascript/src/messages.ts#L685
export enum TestStepResultStatus {
  UNKNOWN = 'UNKNOWN',
  PASSED = 'PASSED',
  SKIPPED = 'SKIPPED',
  PENDING = 'PENDING',
  UNDEFINED = 'UNDEFINED',
  AMBIGUOUS = 'AMBIGUOUS',
  FAILED = 'FAILED',
}

// instrumentation.ts
      [
        TestStepResultStatus.UNDEFINED,
        TestStepResultStatus.AMBIGUOUS,
        TestStepResultStatus.FAILED,
      ].includes(status)
    ) {
```

Then we still preserve the context for future readers, but don't depend on the package.

Reply:
And another option, which might even be better (will not require maintaining a copied enum or magical string list)

`@cucumber/cucumber` is actually exporting this enum in it's `moduleExports` [here](https://github.com/cucumber/cucumber-js/blob/e88c691444b8d88bca67bd464ac30fd89a981b8b/src/index.ts#L56):

```js
export const Status = messages.TestStepResultStatus
```

So you could just pull it from the `patch`, something like:
```
      new InstrumentationNodeModuleDefinition<Cucumber>(
        '@cucumber/cucumber',
        ['^8.0.0'],
        (moduleExports, moduleVersion) => {
          this._diag.debug(
            `Applying patch for @cucumber/cucumber@${moduleVersion}`
          );
          const TestResultStatusEnum = moduleExports.Status; // I think this will work
          steps.forEach(step => {
            this._wrap(moduleExports, hook, this._getHookPatch(hook, TestResultStatusEnum));
          }
```

What you did is already great, these are optional improvements only if you like it.

Reply:
Because the Status is exported from `@cucumber/cucumber`, but I'm using it while instrumenting the `@cucumber/cucumber/lib/runtime/test_case_runner.js` file, I went for this: https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1252/commits/b757c1a4befdc764ce617b7d8246b3b114cfc98e

What do you think?

Reply:
Bummer :/

I would recommend against it:
1. The instrumentation could potentially patch multiple different versions of the instrumented package (@cucumber/cucumber), even though I don't think it's very likely to happen with this specific package. If that happens, then they override each other.
2. It is hard to assert the order in which modules are loaded and patched.

It looks good to me in this specific case, In general - keeping a private class reference to the `moduleExports` from the patch function is considered an anti-pattern for instrumentation and could open the door for bugs that are very hard to find. We had issues in the past with this technic and had to revert.

I recommend copying the enum (but am also good with keeping what you already implemented if you prefer).